src/NHSD.BuyingCatalogue.Identity.Api/Controllers/ConsentController.cs

```diff
@@ -46,5 +52,17 @@ public async Task<IActionResult> Index(ConsentViewModel model)
 
             return View("Error");
         }
+
+        [AllowAnonymous]
+        [HttpGet("/dismiss-cookie-banner")]
+        public IActionResult DismissCookieBanner()
+        {
+            Response.Cookies.Append(Cookies.BuyingCatalogueConsent, "true", new CookieOptions
+            {
+                Expires = DateTime.Now.Add(cookieExpiration.ConsentExpiration),
+            });
+
+            return Redirect(Request.GetTypedHeaders().Referer.ToString());
+        }
```

Comment:
The `ConsentController` was originally for handling consent in the context of identity/IdentityServer only. Ideally, I feel this method probably belongs in a separate `CookieConsent` controller. However, given that this application will soon be retired due to the architectural revamp I think we may as well leave this here.